lib/improver/nowcasting/optical_flow.py

```diff
@@ -297,71 +297,58 @@ class OpticalFlow(object):
     from time-separated fields using an optical flow algorithm
     """
 
-    def __init__(self, data_smoothing_radius_km=7.,
-                 data_smoothing_method='box', boxsize_km=30.,
-                 point_weight=0.1, iterations=100):
+    def __init__(self, data_smoothing_method='box', iterations=100):
         """
         Initialise the class with smoothing parameters for estimating gridded
         u- and v- velocities via optical flow.
 
         Keyword Args:
-            data_smoothing_radius_km (float):
-                Kernel radius in km over which to smooth input data before
-                estimating partial derivatives.  Should not be less than 3x
-                the grid length of the input data cube.
             data_smoothing_method (str):
                 Smoothing method to be used on input fields before estimating
                 partial derivatives.  Can be square 'box' (as used in STEPS) or
                 circular 'kernel' (used in post-calculation smoothing).
-            boxsize_km (float):
-                Square box size in km over which all data points are assumed
-                to have the same velocity to enable matrix inversion
-                (solve_for_uv()).  Should not be less than 3x the grid length
-                of the input data cube.  Must be larger than the kernel radius
-                used to smooth the input data fields.
-            point_weight (float):
-                Weight given to the velocity of a point (box), as opposed to
-                its neighbours, when doing the smart smoothing after velocity
-                calculation.  Values 0-1.
             iterations (int):
                 Number of iterations to perform in post-calculation smoothing.
+                The value for good convergence is 20 (Bowler et al. 2004 [1]).
+
+        Raises:
+            ValueError:
+                If iterations < 20
+
+        [1] Bowler, N., Pierce, C. and Seed, A. 2004: Development of a
+        precipitation nowcasting algorithm based upon optical flow techniques.
+        Journal of Hydrology, 288, 74-91.
 
-        Martina's TODO: need to calculate a suitable kernel size given dt (what
-        is expected movement between slices) and expected velocities
         """
 
-        # parameters for input data smoothing
-        self.data_smoothing_radius_km = data_smoothing_radius_km
+        if iterations < 20:
+            raise ValueError('Minimum requirement 20 iterations')
+
+        # Set parameters for input data smoothing.  14 km is suitable for input
+        # fields separated by a 15 minute time step - this is updated if
+        # necessary by the "process" function.
+        self.data_smoothing_radius_km = 14.
```

Comment:
Is there much point defining `self.data_smoothing_radius` in this `__init__` given that it isn't actually calculated until inside the process method, given that you need to know `grid_length_km` for it to be calculated?  There doesn't seem to be much gained by putting `data_smoothing_radius` on the class, given that the variable is mostly defined and used within the process method.

Reply:
It's used in `process` and `_process_dimensionless`.  Can remove it from `__init__` though.

Reply:
Now updated to have `data_smoothing_radius` as a variable rather than a class member.